lib/common/strings.c

```diff
@@ -1107,6 +1107,11 @@ pcmk__strcmp(const char *s1, const char *s2, uint32_t flags)
         int regcomp_rc = 0;
         int rc = 0;
 
+        if (r_patt == NULL) {
+            crm_err("Out of memory when allocating regex");
+            return 1;
+        }
```

Comment:
Typically we do either CRM_ASSERT(r_patt != NULL) or something like CRM_CHECK(r_patt != NULL, return 1) for memory failures. I'm not sure that's the ideal handling but it works for our purposes, and being out of memory is messy in any case.

However there's no reason to dynamically allocate regex_t. We can just declare r_patt as a regex_t and pass its address to the regex functions.

Reply:
Yeah, definitely.  Dropped the test for if calloc fails, since we no longer call calloc, and updated the code to use a plain old variable.